libraries/standard.js

```diff
@@ -1,8 +1,32 @@
-const lib = new Map();
+// Description: Parse a CSV string into an array of objects
 
-// const example_function = function(example_parameter) {
-//     console.log("some example behaviour " + example_parameter);
-// }
-// lib.set("example_function", example_function);
+const csv_parser = function (csv_string) { 
+    // Source: https://stackoverflow.com/questions/8493195/how-can-i-parse-a-csv-string-with-javascript-which-contains-comma-in-data
+    var re_valid = /^\s*(?:'[^'\\]*(?:\\[\S\s][^'\\]*)*'|"[^"\\]*(?:\\[\S\s][^"\\]*)*"|[^,'"\s\\]*(?:\s+[^,'"\s\\]+)*)\s*(?:,\s*(?:'[^'\\]*(?:\\[\S\s][^'\\]*)*'|"[^"\\]*(?:\\[\S\s][^"\\]*)*"|[^,'"\s\\]*(?:\s+[^,'"\s\\]+)*)\s*)*$/;
+    
+    if (!re_valid.test(csv_string)) {
+        console.log("Invalid CSV string ")
+        return null;
+    }
 
-return lib;
\ No newline at end of file
+    let new_data = [];
+    let lines = csv_string.split(/\r|\n/); // Current delimiters is return or newline
+    let headers = lines[0].split(",");
+    lines.shift();
+
+    lines.map(d => {
+        let obj = {};
+        let currentline = d.split(",");
+        headers.map((h, i) => {
+            let temp_num = Number(currentline[i]);
+            if (temp_num) {
+                obj[h] = temp_num;
+            } else {
+                obj[h] = currentline[i];
+            }
+        })
+        new_data.push(obj);
+    })
+    return new_data;
+}
+return csv_parser;
```

Comment:
this should be `lib.set("csv_parser", csv_parser);` followed by `return lib;` as mentioned in the comment above